### DISCUSSION TITLE
Доступ к таблицам в цикле

### DISCUSSION THREAD

**Discussion by tvideo** [totumonline/totum-ru-issues-and-discussions#449] [Как это сделать?] [answered]
Здравствуйте. 
Подскажите, пожалуйста, как решить следующую задачку:
Есть таблица циклов, в каждом цикле 2 таблицы расчетные в цикле. Нужно что бы одна группа пользователей имела доступ только к одной из них, ко второй. При запрете первой расчетной в цикле, пользователь из заданной группы при создании нового цикла переносится сразу в первую расчетную в цикле, к которой доступ запрещен. Как поменять их местами, что бы при создании нового цикла пользователем из заданной группы, он перемещался сразу во вторую таблицу "РАСЧЕТ СТОИМОСТИ"

![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/32614108/3127f521-c753-4c8d-b80e-23e52497c8db)

Спасибо

**Comment by erpic-ru** [comment, marked as answer]
Можно закастомизировать кнопку добавления. Все как в примере, только в зависимости от роли/пользователя подсовывать нужную таблицу для перехода:
https://www.youtube.com/watch?v=950C-q44REM&t=72s

**Comment by tvideo** [comment]
ок, спасибо)
Может еще подскажите как кодом добавить несколько фалов в одно поле "файл" с множественным значением?)

**Reply by erpic-ru** [reply]
При помощи listreplace перебрать список  ссылок и собрать массив в поле с файлами

Что-то типа такого: 
```
= : listReplace(list: $links; action: "filestringbase64" = $base; action: "name" = $names value: "value")

names: список имен
links: - список ссылок

base: strBaseEncode(str: $file)
			file: getFromScript(uri: $#value)
```

**Reply by tvideo** [reply]
ок. Спасибо
